src/middleware/github-webhook-middleware.ts

```diff
@@ -71,7 +71,7 @@ function extractWebhookEventNameFromContext(context: WebhookContext): string {
 
 // TODO: fix typings
 export const GithubWebhookMiddleware = (
-	callback: (webhookContext: WebhookContext, jiraClient: any, util: any, githubInstallationId: number) => Promise<void>
+	callback: (webhookContext: WebhookContext, jiraClient: any, util: any, githubInstallationId: number, subscription: Subscription) => Promise<void>
```

Comment:
Passing in the subscription, because we need it in the push webhook handler to find the `RepoSyncState` for the given subscription.

Reply:
Do we really need to pass `Subscription`? It can be fetched using `githubInstallationId` and `githubAppId` in push webhook handler if config file has been changed.

Reply:
Why fetch it again if it's already available here? This way saves one database call.

Reply:
Yep, it is saving one DB call but only in one case when config file is created/updated which is going to be very rare.  The only point I'm trying to emphasise is why to have pass another argument (already having 4 arguments) that makes code more complex without having any significant advantage.
This is small issue, you can ignore it.